### ISSUE TITLE
Signaling: Subsequent responder handshakes trigger a protocol error

### ISSUE THREAD

**Issue by ovalseven8** [threema-ch/threema-web#799]
Using Threema Web's beta, I encountered the following problem today:
![Bildschirmfoto vom 2019-05-10 19-40-23](https://user-images.githubusercontent.com/8258609/57545750-9e814180-735b-11e9-9a59-ca64fac5a236.png)

After the reconnection did not work, Threema Web showed this screen (note that the password is still kind of displayed but I cannot edit it, the input field is disabled).

![Screenshot_2019-05-10 Threema Web](https://user-images.githubusercontent.com/8258609/57545864-da1c0b80-735b-11e9-9d18-002eb8b9e740.png)

<details>
<summary>Full logs</summary>

```
[StateService] Reset states angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
Detected browser: Firefox 66 angular.js:15536:16
[SettingsService] Removing settings key: v2infoShown angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
[StateService] Reset states angular.js:15536:16
[WelcomeController] Initialize session by unlocking trusted key... angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<input id="input_0" class="ng-untouched ng-valid md…ng-dirty ng-valid-parse" type="password" ng-model="ctrl.password" ng-disabled="ctrl.formLocked" autofocus="" aria-labelledby="aria-label-password-reconnect" translate-attr="{'placeholder': 'welcome.PASSWORD'}" autocomplete="current-password" aria-invalid="false" placeholder="Passwort" style="">
angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<button class="md-raised md-primary md-… ng-scope md-ink-ripple" type="submit" ng-transclude="" translate="" translate-attr-aria-label="welcome.BTN_RECONNECT" aria-label="Verbindung aufbauen">
angular.js:15536:16
[VersionService] Using Threema Web version 2.1.7-reconnect-overhaul-beta6 angular.js:15536:16
[WelcomeController] Checking if the session is already open in another tab or window angular.js:15536:16
[WebClientService] Stopping (reason=stop, send=false, close=welcome, connectionBuildupState=new) angular.js:15536:16
[WebClientService] Timer stopped angular.js:15536:16
[StateService] Reset states angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
[TimeoutService] Cancelling 0 timeouts angular.js:15536:16
[WebClientService] Session closed (cannot be resumed) angular.js:15536:16
[WebClientService] Peer connection was null angular.js:15536:16
[PushService] Initialized with gcm token angular.js:15536:16
Initializing (keyStore=yes, peerTrustedKey=yes, resume=false) angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
[TimeoutService] Cancelling 0 timeouts angular.js:15536:16
Discarded previous session angular.js:15536:16
[WebClientService] Starting WebClientService... angular.js:15536:16
[WebClientService] Creating new startup promise angular.js:15536:16
[StateService] Connection buildup state: new => push angular.js:15536:16
[StateService] Signaling connection state: new => new angular.js:15536:16
[StateService] Signaling connection state: new => ws-connecting angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Sending push 1/3 (ttl=0) angular.js:15536:16
[StatusController] State change: error -> warning angular.js:15536:16
[StateService] Signaling connection state: ws-connecting => server-handshake angular.js:15536:16
[StateService] Signaling connection state: server-handshake => peer-handshake angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push sent successfully angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Sending push 2/3 (ttl=15) angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push sent successfully angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Sending push 3/3 (ttl=90) angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push sent successfully angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push done angular.js:15536:16
[StateService] Connection buildup state: push => peer_handshake angular.js:15536:16
[PeerConnectionHelper] Initialize WebRTC PeerConnection angular.js:15536:16
[PeerConnectionHelper] ICE servers used: turn:ds-turn.threema.ch:443?transport=udp, turn:ds-turn.threema.ch:443?transport=tcp, turns:ds-turn.threema.ch:443 angular.js:15536:16
Using more than two STUN/TURN servers slows down discovery app.bundle.js:60
[PeerConnectionHelper] Setting up ICE candidate handling angular.js:15536:16
[StateService] Signaling connection state: peer-handshake => task angular.js:15536:16
[PeerConnectionHelper] RTCPeerConnection: negotiation needed angular.js:15536:16
[SaltyRTC.Initiator] Signaling error: Protocol error (Could not find peer 3) saltyrtc-client.es5.js:1357:25
[PeerConnectionHelper] Signaling state change: have-local-offer angular.js:15536:16
[PeerConnectionHelper] Created offer, set local description angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send offer: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:783:13
[StateService] Connection buildup state: peer_handshake => closed angular.js:15536:16
[StateService] Ignored signaling connection state to "closed" angular.js:15536:16
[StateService] Connection buildup state: closed => connecting angular.js:15536:16
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<input id="input_1" class="ng-pristine ng-untouched…d md-input ng-not-empty" type="password" ng-model="ctrl.password" ng-disabled="ctrl.formLocked" autofocus="" aria-labelledby="aria-label-password-reconnect" translate-attr="{'placeholder': 'welcome.PASSWORD'}" autocomplete="current-password" disabled="disabled" aria-invalid="false" placeholder="Passwort">
angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<button class="md-raised md-primary md-… ng-scope md-ink-ripple" type="submit" ng-transclude="" translate="" translate-attr-aria-label="welcome.BTN_RECONNECT" aria-label="Verbindung aufbauen">
angular.js:15536:16
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[PeerConnectionHelper] ICE gathering state change: gathering angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:0 1 UDP 2122187007 192.168.178.34 1 typ host angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:3 1 UDP 2122252543 2003:d8:73c7:a100:e758:f49c:5bb7:2809 1 typ host angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:6 1 TCP 2105458943 192.168.178.34 1 typ host tcptype active angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:9 1 TCP 2105524479 2003:d8:73c7:a100:e758:f49c:5bb7:2809 1 typ host tcptype active angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send candidates: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:833:15
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
Source-Map-Fehler: request failed with status 404
Ressourcen-Adresse: https://web-beta.threema.ch/dist/app.bundle.js?v=2.1.7-reconnect-overhaul-beta6
Source-Map-Adresse: app.bundle.js.map[Weitere Informationen]
[PeerConnectionHelper] Gathered local ICE candidate: candidate:1 1 UDP 1685987327 91.46.16.167 1 typ srflx raddr *** rport 2 angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:2 1 UDP 92151807 185.88.236.77 59282 typ relay raddr *** rport 2 angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send candidates: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:833:15
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:5 1 UDP 92217343 185.88.236.77 57572 typ relay raddr *** rport 2 angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send candidates: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:833:15
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:7 1 UDP 8265727 185.88.236.77 64095 typ relay raddr *** rport 2 angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send candidates: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:833:15
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:10 1 UDP 8331263 185.88.236.77 60550 typ relay raddr *** rport 2 angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send candidates: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:833:15
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:11 1 UDP 8331263 212.103.68.7 54646 typ relay raddr *** rport 2 angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send candidates: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:833:15
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[PeerConnectionHelper] ICE gathering state change: complete angular.js:15536:16
[PeerConnectionHelper] No more local ICE candidates angular.js:15536:16
Ein iframe, der sowohl "allow-scripts" als auch "allow-same-origin" für sein "sandbox"-Attribut gesetzt hat, kann seine Sandbox entfernen. web-beta.threema.ch
Source-Map-Fehler: request failed with status 404
Ressourcen-Adresse: https://web-beta.threema.ch/dist/app.bundle.js?v=2.1.7-reconnect-overhaul-beta6
Source-Map-Adresse: app.bundle.js.map[Weitere Informationen]

```
</details>

## Your Environment
* Threema Web version: 2.1.7-reconnect-overhaul-beta6
* Threema App version: 3.7 (Android)
* Browser name and version: Firefox 66.0.5

**Comment by ovalseven8** [comment]
Some more information:

Today, I tried to reconnect (using web-beta.threema.ch) and the 12 % issue occurred again (also see https://github.com/threema-ch/threema-web/pull/792#issuecomment-491349519). However, after about 3 minutes, the 12 % disappeared and the display above appeared (with the disabled password input). When I look into the logs, the app definitely got the push notification.

Here detailed logs. At 13:43:25 I clicked "Verbindung aufbauen" on web-beta.threema.ch:

<details>
<summary>Logs from web-beta.threema.ch</summary>

```
[StateService] Reset states angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
Detected browser: Firefox 66 angular.js:15536:16
[SettingsService] Removing settings key: v2infoShown angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
[StateService] Reset states angular.js:15536:16
[WelcomeController] Initialize session by unlocking trusted key... angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<input id="input_0" class="ng-untouched ng-valid md…ng-dirty ng-valid-parse" type="password" ng-model="ctrl.password" ng-disabled="ctrl.formLocked" autofocus="" aria-labelledby="aria-label-password-reconnect" translate-attr="{'placeholder': 'welcome.PASSWORD'}" autocomplete="current-password" aria-invalid="false" placeholder="Passwort" style="">
angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<button class="md-raised md-primary md-… ng-scope md-ink-ripple" type="submit" ng-transclude="" translate="" translate-attr-aria-label="welcome.BTN_RECONNECT" aria-label="Verbindung aufbauen">
angular.js:15536:16
[VersionService] Using Threema Web version 2.1.7-reconnect-overhaul-beta6 angular.js:15536:16
[WelcomeController] Checking if the session is already open in another tab or window angular.js:15536:16
[WebClientService] Stopping (reason=stop, send=false, close=welcome, connectionBuildupState=new) angular.js:15536:16
[WebClientService] Timer stopped angular.js:15536:16
[StateService] Reset states angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
[TimeoutService] Cancelling 0 timeouts angular.js:15536:16
[WebClientService] Session closed (cannot be resumed) angular.js:15536:16
[WebClientService] Peer connection was null angular.js:15536:16
[PushService] Initialized with gcm token angular.js:15536:16
Initializing (keyStore=yes, peerTrustedKey=yes, resume=false) angular.js:15536:16
[WebClientService] Reset initialization steps angular.js:15536:16
[TimeoutService] Cancelling 0 timeouts angular.js:15536:16
Discarded previous session angular.js:15536:16
[WebClientService] Starting WebClientService... angular.js:15536:16
[WebClientService] Creating new startup promise angular.js:15536:16
[StateService] Connection buildup state: new => push angular.js:15536:16
[StateService] Signaling connection state: new => new angular.js:15536:16
[StateService] Signaling connection state: new => ws-connecting angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Sending push 1/3 (ttl=0) angular.js:15536:16
[StatusController] State change: error -> warning angular.js:15536:16
[StateService] Signaling connection state: ws-connecting => server-handshake angular.js:15536:16
[StateService] Signaling connection state: server-handshake => peer-handshake angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push sent successfully angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Sending push 2/3 (ttl=15) angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push sent successfully angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Sending push 3/3 (ttl=90) angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push sent successfully angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push session timeout after 3 tries angular.js:15536:16
[Push.56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a] Push done angular.js:15536:16
[StateService] Connection buildup state: push => peer_handshake angular.js:15536:16
[PeerConnectionHelper] Initialize WebRTC PeerConnection angular.js:15536:16
[PeerConnectionHelper] ICE servers used: turn:ds-turn.threema.ch:443?transport=udp, turn:ds-turn.threema.ch:443?transport=tcp, turns:ds-turn.threema.ch:443 angular.js:15536:16
[PeerConnectionHelper] Setting up ICE candidate handling angular.js:15536:16
Using more than two STUN/TURN servers slows down discovery app.bundle.js:60
[PeerConnectionHelper] RTCPeerConnection: negotiation needed angular.js:15536:16
[SaltyRTC.Initiator] Signaling error: Protocol error (Could not find peer 3) saltyrtc-client.es5.js:1357:25
[PeerConnectionHelper] Signaling state change: have-local-offer angular.js:15536:16
[PeerConnectionHelper] Created offer, set local description angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send offer: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:783:13
[PeerConnectionHelper] ICE gathering state change: gathering angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:0 1 UDP 2122187007 192.168.178.34 1 typ host angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:3 1 UDP 2122252543 2003:d8:73c7:a100:e758:f49c:5bb7:2809 1 typ host angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:6 1 TCP 2105458943 192.168.178.34 1 typ host tcptype active angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:9 1 TCP 2105524479 2003:d8:73c7:a100:e758:f49c:5bb7:2809 1 typ host tcptype active angular.js:15536:16
Source-Map-Fehler: request failed with status 404
Ressourcen-Adresse: https://web-beta.threema.ch/dist/app.bundle.js?v=2.1.7-reconnect-overhaul-beta6
Source-Map-Adresse: app.bundle.js.map[Weitere Informationen]
[PeerConnectionHelper] Gathered local ICE candidate: candidate:5 1 UDP 92217343 212.103.68.7 59633 typ relay raddr *** rport 2 angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:11 1 UDP 8331263 185.88.236.76 54631 typ relay raddr *** rport 2 angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:10 1 UDP 8331263 212.103.68.7 60823 typ relay raddr *** rport 2 angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:7 1 UDP 8265727 212.103.68.7 56588 typ relay raddr *** rport 2 angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:1 1 UDP 1685987327 91.46.16.167 1 typ srflx raddr *** rport 2 angular.js:15536:16
[PeerConnectionHelper] Gathered local ICE candidate: candidate:2 1 UDP 92151807 212.103.68.7 56615 typ relay raddr *** rport 2 angular.js:15536:16
[StateService] Signaling connection state: peer-handshake => task angular.js:15536:16
[StateService] Connection buildup state: peer_handshake => closed angular.js:15536:16
[StateService] Ignored signaling connection state to "closed" angular.js:15536:16
[StateService] Connection buildup state: closed => connecting angular.js:15536:16
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<input class="ng-pristine ng-untouched ng-valid" type="password" ng-model="ctrl.password" ng-disabled="ctrl.formLocked" autofocus="" aria-labelledby="aria-label-password-reconnect" translate-attr="{'placeholder': 'welcome.PASSWORD'}" autocomplete="current-password">
angular.js:15536:16
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: 
<button class="md-raised md-primary md-… ng-scope md-ink-ripple" type="submit" ng-transclude="" translate="" translate-attr-aria-label="welcome.BTN_RECONNECT">
angular.js:15536:16
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[SaltyRTC.WebRTC.initiator] Could not send candidates: Cannot send task message in "new" state saltyrtc-task-webrtc.es5.js:833:15
[StateService] Ignored signaling connection state to "new" angular.js:15536:16
[PeerConnectionHelper] ICE gathering state change: complete angular.js:15536:16
[PeerConnectionHelper] No more local ICE candidates angular.js:15536:16
```
</details>

<details>
<summary>message_log.txt</summary>

```
Sun May 12 13:43:26 GMT+02:00 2019	Handling incoming GCM intent.
Sun May 12 13:43:26 GMT+02:00 2019	Received GCM message: Bundle[{google.delivered_priority=high, google.sent_time=1557661405799, google.original_priority=high, wca=TOqcDE, wcs=56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a, wct=1557661405, wcv=2, google.message_id=0:1557661405801529%f5f551dcf9fd7ecd}]
Sun May 12 13:43:26 GMT+02:00 2019	*** Message sent     : Sun May 12 13:43:25 GMT+02:00 2019
Sun May 12 13:43:26 GMT+02:00 2019	*** Message received : Sun May 12 13:43:26 GMT+02:00 2019
Sun May 12 13:43:26 GMT+02:00 2019	GcmMessageListenerService: Received GCM webclient wakeup for session 56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a
Sun May 12 13:43:26 GMT+02:00 2019	GcmMessageListenerService: Trying to wake up webclient session 56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a
Sun May 12 13:43:26 GMT+02:00 2019	Webclient Session Wakeup: startWakeUp
Sun May 12 13:43:26 GMT+02:00 2019	Webclient Session Wakeup: start
Sun May 12 13:43:26 GMT+02:00 2019	Webclient Session Wakeup: Getting SessionInstanceService
Sun May 12 13:43:28 GMT+02:00 2019	SessionInstanceService [0]: Initialize SessionInstanceServiceImpl
Sun May 12 13:43:28 GMT+02:00 2019	Handling incoming GCM intent.
Sun May 12 13:43:28 GMT+02:00 2019	Received GCM message: Bundle[{google.delivered_priority=high, google.sent_time=1557661408004, google.ttl=15, google.original_priority=high, wca=TOqcDE, wcs=56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a, wct=1557661407, wcv=2, google.message_id=0:1557661408008000%f5f551dc60a376ef, collapse_key=relay.56a299}]
Sun May 12 13:43:28 GMT+02:00 2019	*** Message sent     : Sun May 12 13:43:28 GMT+02:00 2019
Sun May 12 13:43:28 GMT+02:00 2019	*** Message received : Sun May 12 13:43:28 GMT+02:00 2019
Sun May 12 13:43:28 GMT+02:00 2019	GcmMessageListenerService: Received GCM webclient wakeup for session 56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a
Sun May 12 13:43:28 GMT+02:00 2019	GcmMessageListenerService: Trying to wake up webclient session 56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a
Sun May 12 13:43:28 GMT+02:00 2019	Webclient Session Wakeup: startWakeUp
Sun May 12 13:43:28 GMT+02:00 2019	Webclient Session Wakeup: start
Sun May 12 13:43:28 GMT+02:00 2019	Webclient Session Wakeup: Getting SessionInstanceService
Sun May 12 13:43:28 GMT+02:00 2019	Webclient Session Wakeup: Resuming SessionInstanceService
Sun May 12 13:43:28 GMT+02:00 2019	SessionInstanceService [0]: Resuming Threema Web session
Sun May 12 13:43:28 GMT+02:00 2019	Webclient Session Wakeup: Resuming SessionInstanceService
Sun May 12 13:43:28 GMT+02:00 2019	SessionInstanceService [0]: Resuming Threema Web session
Sun May 12 13:43:32 GMT+02:00 2019	Handling incoming GCM intent.
Sun May 12 13:43:32 GMT+02:00 2019	Received GCM message: Bundle[{google.delivered_priority=high, google.sent_time=1557661412155, google.ttl=90, google.original_priority=high, wca=TOqcDE, wcs=56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a, wct=1557661412, wcv=2, google.message_id=0:1557661412157658%f5f551dc60a376ef, collapse_key=relay.56a299}]
Sun May 12 13:43:32 GMT+02:00 2019	*** Message sent     : Sun May 12 13:43:32 GMT+02:00 2019
Sun May 12 13:43:32 GMT+02:00 2019	*** Message received : Sun May 12 13:43:32 GMT+02:00 2019
Sun May 12 13:43:32 GMT+02:00 2019	GcmMessageListenerService: Received GCM webclient wakeup for session 56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a
Sun May 12 13:43:32 GMT+02:00 2019	GcmMessageListenerService: Trying to wake up webclient session 56a299e0b5dc5d7a095de65a1cf3b6fc5ea0cd8351d84b538ba4360b17a6f60a
Sun May 12 13:43:32 GMT+02:00 2019	Webclient Session Wakeup: startWakeUp
Sun May 12 13:43:32 GMT+02:00 2019	Webclient Session Wakeup: start
Sun May 12 13:43:32 GMT+02:00 2019	Webclient Session Wakeup: Getting SessionInstanceService
Sun May 12 13:43:32 GMT+02:00 2019	Webclient Session Wakeup: Resuming SessionInstanceService
Sun May 12 13:43:32 GMT+02:00 2019	SessionInstanceService [0]: Resuming Threema Web session
Sun May 12 13:45:24 GMT+02:00 2019	SessionStateManager[0]: Session state changed from NEW to CONNECTING
Sun May 12 13:45:24 GMT+02:00 2019	SessionStateManager[0]: Session state changed from CONNECTING to CONNECTING
Sun May 12 13:45:24 GMT+02:00 2019	Webclient Session Wakeup: Webclient session started
Sun May 12 13:45:24 GMT+02:00 2019	SessionStateManager[0]: Session state changed from CONNECTING to CONNECTING
Sun May 12 13:45:24 GMT+02:00 2019	Webclient Session Wakeup: Webclient session started
Sun May 12 13:45:24 GMT+02:00 2019	Webclient Session Wakeup: Webclient session started
Sun May 12 13:45:24 GMT+02:00 2019	ch.threema.app.ThreemaApplication: WebClientListenerManager: onStateChanged
Sun May 12 13:45:24 GMT+02:00 2019	ch.threema.app.ThreemaApplication: WebClientListenerManager: onStateChanged
Sun May 12 13:45:24 GMT+02:00 2019	ch.threema.app.ThreemaApplication: WebClientListenerManager: onStateChanged
Sun May 12 13:45:25 GMT+02:00 2019	SessionStateManager[0]: Session state changed from CONNECTING to DISCONNECTED
Sun May 12 13:45:25 GMT+02:00 2019	SessionState[0,DISCONNECTED]: Cleanup
Sun May 12 13:45:25 GMT+02:00 2019	SessionState[0,DISCONNECTED]: Cleanup done
Sun May 12 13:45:25 GMT+02:00 2019	ch.threema.app.ThreemaApplication: WebClientListenerManager: onStopped
Sun May 12 13:45:25 GMT+02:00 2019	ch.threema.app.ThreemaApplication: WebClientListenerManager: onStateChanged
Sun May 12 13:45:25 GMT+02:00 2019	SessionInstanceService [0]: Stopping session instance: DisconnectContext{requestedByUs=false, reason=null}
Sun May 12 13:45:25 GMT+02:00 2019	ch.threema.app.ThreemaApplication: SessionAndroidService not running...not stopping
Sun May 12 13:45:25 GMT+02:00 2019	ch.threema.app.ThreemaApplication: updating SessionAndroidService
Sun May 12 13:45:25 GMT+02:00 2019	ch.threema.app.ThreemaApplication: SessionAndroidService not running...not updating
Sun May 12 13:48:36 GMT+02:00 2019	SessionState[0,DISCONNECTED]: Error: Timeout while connecting to remote client
Sun May 12 13:48:36 GMT+02:00 2019	SessionStateManager[0]: Session state changed from DISCONNECTED to ERROR
Sun May 12 13:48:36 GMT+02:00 2019	SessionState[0,ERROR]: Error: Timeout while connecting to remote client
Sun May 12 13:48:36 GMT+02:00 2019	ch.threema.app.ThreemaApplication: WebClientListenerManager: onStateChanged
```
</details>

**Comment by lgrahl** [comment]
This is a bug in the SaltyRTC Client JS lib: https://github.com/saltyrtc/saltyrtc-client-js/issues/114

@dbrgn please take a look at the second comment. It looks to me as if the app establishes multiple SaltyRTC connections but dispatches to a single, central state.

**Comment by lgrahl** [comment]
Should have been resolved by 014a05a
